### ISSUE TITLE
Ability to add multiple TODO across the code, but refering to the same issue?

### ISSUE THREAD

**Issue by naps62** [alstr/todo-to-issue-action#215]
I know we can use `TODO(#99)`, but that requires the issue number to already exist.

In my use cases, I often have to add a TODO in multiple parts of the codebase, but with it being actually the same overall task (e.g. some missing feature that, once implemented, will require my attention in various points of the code).

Is it possible to somehow have all these comments (written during the same PR) end up creating a single issue?

**Comment by alstr** [comment]
Not at this stage. You'd have to create the issue first, then you could use that issue number in each TODO to add to it.

The challenge when when the issue doesn't already exist is how to tie separate code blocks together. There'd have to be some sort of common identifier. It's doable, but I'm not sure what that would be.

**Comment by rgalonso** [comment]
Perhaps a way to handle this would be to support a placeholder reference. e.g.
```diff
+ // TODO(new): fix the widget
```
All references to `new` would be replaced with the same new issue number.

Like for [URL Insertion](https://github.com/alstr/todo-to-issue-action?tab=readme-ov-file#url-insertion), this would require that the workflow include steps to set the Git user and commit/push changes.

This could be extended to support multiple new issues in the same run.

```diff
+ // TODO(new-1): add function documentation
int transform(int input) {
-   in output = input*20 + 5;
+   int output = input*20 + 5; // TODO(new-2): replace magical numbers
  return output;
}

+ // TODO(new-1): add function documentation
int inverseTransform(int input) {
-   int output = (input - 5)/20;
+   int output = (input - 5)/20; // TODO(new-2): replace magical numbers
  return output;
}
```

(`new` and `new-1` should be treated equivalently.)

What do you think of this approach?

-----

There would have to be some rules/checks about the issue title. i.e. what if in the last changed line above they had written it differently, or simply had a typo? (e.g. `replace magcial numbres`) Which is the "right" title to use? There are likely other corner cases that would need to be considered, but I wanted to record that one since it already occurred to me now.

**Comment by alstr** [comment]
Yes, that's a good idea, maybe with some kind of unambiguous symbol prepended, to maintain consistency with the other functionality, e.g. `TODO(*new)`.

I suppose it comes down to how useful/widely used it would be vs the complexity to implement. I can see the value in it.

**Comment by alstr** [comment]
Do you think subissues would work well with this?

**Comment by rgalonso** [comment]
> Do you think subissues would work well with this?

Sub-issues may be a feature to add in the future, but I see that as different from this. But while we're discussing it, a way to accomplish that may be syntax like `TODO(new-1): xyz`, `TODO(new-1.1: xyz subissue`, `TODO(new-1.2): xyz different subissue`.

But as I said, I see that as separate so adding support for sub-issues shouldn't hold up this issue, IMO.
